torch/_inductor/fx_passes/joint_graph.py

```diff
@@ -112,6 +112,65 @@ def replace_no_op(node, replace_input_index):
             replace_no_op(node, 0)
 
 
+@torch.utils._python_dispatch._disable_current_modes()
+def remove_redundant_complex_views(gm: torch.fx.GraphModule):
+    """
+    Removes redundant complex views by reusing existing ones.
+    """
+
+    # A dictionary mapping a tensor to all aliased views.
+    aliases = {}
+    graph = gm.graph
+
+    for node in graph.nodes:
+        if node.op != "call_function":
+            continue
+
+        if node.target != torch.ops.aten.view.dtype:
+            continue
+
+        src = node.args[0]
+        from_type = src.meta["val"].dtype
+        to_type = node.args[1]
+
+        # Only handle complex views.
+        if not from_type.is_complex and not to_type.is_complex:
+            continue
+
+        existing_aliases = aliases.get(src)
+        is_needed = True
+
+        if existing_aliases:
+            # Replace the view with the an existing view if available.
+            for alias in existing_aliases:
+                if alias.meta["val"].dtype == to_type:
+                    is_needed = False
+                    node.replace_all_uses_with(alias)
+                    alias.meta.update(node.meta)
+                    graph.erase_node(node)
+                    break
+        else:
+            existing_aliases = [src]
+            aliases[src] = existing_aliases
+
+        if is_needed:
+            # Save the new alias.
+            existing_aliases.append(node)
+            aliases[node] = existing_aliases
+
+    # Clean up unused aliases.
+    while True:
+        unused_aliases = []
+        for alias in aliases:
+            if not alias.users:
+                unused_aliases.append(alias)
+        if len(unused_aliases) == 0:
+            break
+        for unused in unused_aliases:
+            aliases.pop(unused)
+            graph.erase_node(unused)
```

Comment:
Don't these already get erased on L150?  Why do we need to do it again?

Reply:
For patterns like

```
c1 = view(f1, complex64)

f2 = view(c1, float32)

f3 = f2 + f4
```

L150 will replace all uses of `f2` with `f1` and remove `f2`. This leaves `c1` unused. The code here removes `c1` and recursively all its predecessors.